Modules/_threadmodule.c

```diff
@@ -711,6 +711,10 @@ PyThreadHandleObject_is_done(PyObject *op, PyObject *Py_UNUSED(dummy))
 {
     PyThreadHandleObject *self = PyThreadHandleObject_CAST(op);
     if (_PyEvent_IsSet(&self->handle->thread_is_exiting)) {
+        if (_PyOnceFlag_CallOnce(&self->handle->once, join_thread, self->handle) == -1) {
+            PyErr_SetString(PyExc_RuntimeError, "failed to join thread");
```

Comment:
We don't need the `PyErr_SetString` because `join_thread` already sets the exception when it returns `-1`.

```suggestion
```